src/Datasource/Paging/NumericPaginator.php

```diff
@@ -267,7 +267,7 @@ protected function getQuery(RepositoryInterface $object, ?QueryInterface $query,
 
         if ($query === null) {
             $args = [];
-            $type = !empty($options['finder']) ? $options['finder'] : 'all';
+            $type = $options['finder'] ?: 'all';
```

Comment:
Is this always declared?as `?:` would throw a notice otherwise right?

Reply:
Righ, it would. The test suite passed so I guess the key is always set. I'll check and ensure once I am at a PC.

Reply:
No its not, i will revert this change

Reply:
?? Can be used to be safe

Reply:
This looks horrible :D `$type = ($options['finder'] ?? 'all') ?: 'all';`

I cannot use `$options['finder'] ?? 'all';` because it would use 'all' only if finder is not defined or is null.

Reply:
`finder` is a default config which is always set so we can simply the statement to just `$type = $options['finder']`.
The empty check was needed because in older Cake versions a default value was not set for this config.